### PR TITLE
ENH: QIIME 2 Format cleanup

### PR DESCRIPTION
**Pull request by thermokarst**
Hey @arivers! The diff here is a bit hairy, but basically this PR attempts to simplify a lot of the filepath handling wrt QIIME 2 formats. I am marking this as a draft for the moment, I have a few more test case to take care of. Thanks!

### CODE REVIEW

**Comment by arivers** [comment]
@thermokarst, this is great, thanks for your help. When you are done I'll take a look, test and merge everything in. 

 I'm planning to do a  refactoring in late April to add support for reverse complemented primers, merge the itsxpress and q2-itsxpress packages into one and remove some complicated and rarely used input options.

File: setup.py [hunk 1]
```diff
@@ -19,7 +19,9 @@
     python_requires=">3.5",
     include_package_data=True,
     install_requires=[
-        'itsxpress>=1.7.0'
+        'itsxpress>=1.7.0',
+        'qiime2',
```
Comment (thermokarst) [inline setup.py:23]: `qiime2` is a concrete dependency.

File: setup.py [hunk 2]
```diff
@@ -19,7 +19,9 @@
...
+        'pandas',
```
Comment (thermokarst) [inline setup.py:24]: Technically not necessary, but, `pandas` is a secondary dependency (via `qiime2`), and it buys us some conveniences when it comes to `.view()`.

File: q2_itsxpress/_itsxpress.py [hunk 3]
```diff
@@ -19,78 +19,42 @@
...
 import shutil
 import math
 
-import yaml
+import pandas as pd
 from q2_types.per_sample_sequences import (SingleLanePerSamplePairedEndFastqDirFmt,
                                            SingleLanePerSampleSingleEndFastqDirFmt,
-                                           FastqManifestFormat,
-                                           YamlFormat)
-from q2_types.per_sample_sequences._format import _SingleLanePerSampleFastqDirFmt
```
Comment (thermokarst) [inline q2_itsxpress/_itsxpress.py:30]: Removing unsafe format usage --- this is a private format in q2-types, not intended for public consumption.

File: q2_itsxpress/_itsxpress.py [hunk 4]
```diff
@@ -19,78 +19,42 @@
...
+                                           CasavaOneEightSingleLanePerSampleDirFmt)
 from itsxpress import main as itsxpress
 from itsxpress.definitions import (taxa_dict,
                    ROOT_DIR)
 
 default_cluster_id=0.995
 
-def _view_artifact_type(per_sample_sequence: _SingleLanePerSampleFastqDirFmt) -> str:
```
Comment (thermokarst) [inline q2_itsxpress/_itsxpress.py:37]: Util no longer necessary with this refactor.

File: q2_itsxpress/_itsxpress.py [hunk 5]
```diff
@@ -107,78 +71,19 @@ def _set_fastqs_and_check(per_sample_sequences: _SingleLanePerSampleFastqDirFmt,
...
-        return sequence_id, sobj
+        return sobj
 
     elif not paired_end and not interleaved:
         sobj = itsxpress.SeqSampleNotPaired(fastq=fastq,
                                             tempdir=None)
-        return sequence_id, sobj
-
-
-def _write_metadata(results: SingleLanePerSampleSingleEndFastqDirFmt):
```
Comment (thermokarst) [inline q2_itsxpress/_itsxpress.py:125]: Util no longer necessary with this refactor.

File: q2_itsxpress/_itsxpress.py [hunk 6]
```diff
@@ -107,78 +71,19 @@ def _set_fastqs_and_check(per_sample_sequences: _SingleLanePerSampleFastqDirFmt,
...
-        of the output.
-
-    """
-
-    metadata = YamlFormat()
-    metadata.path.write_text(yaml.dump({'phred-offset': 33}))
-    results.metadata.write_data(metadata, YamlFormat)
-
-
-def _fastq_id_maker(per_sample_sequences: _SingleLanePerSampleFastqDirFmt,
```
Comment (thermokarst) [inline q2_itsxpress/_itsxpress.py:139]: Util no longer necessary with this refactor.

File: q2_itsxpress/_itsxpress.py [hunk 7]
```diff
@@ -204,12 +109,13 @@ def trim_single(per_sample_sequences: SingleLanePerSampleSingleEndFastqDirFmt,
                 region: str,
                 taxa: str = "F",
                 threads: int = 1,
-                cluster_id: float = default_cluster_id) -> SingleLanePerSampleSingleEndFastqDirFmt:
+                cluster_id: float = default_cluster_id) -> CasavaOneEightSingleLanePerSampleDirFmt:
```
Comment (thermokarst) [inline q2_itsxpress/_itsxpress.py:112]: `CasavaOneEightSingleLanePerSampleDirFmt` is a more generic container format, accepting both single and paired reads. This affords us some flexibility!

File: q2_itsxpress/_itsxpress.py [hunk 8]
```diff
@@ -233,67 +140,57 @@ def trim_pair_output_unmerged(per_sample_sequences: SingleLanePerSamplePairedEnd
...
     taxa = _taxa_prefix_to_taxa(taxa)
-    # Writing the manifest for the output qza
-    manifest = FastqManifestFormat()
-    manifest_fn = manifest.open()
-    manifest_fn.write('sample-id,filename,direction\n')
-    # Getting the sequences from the manifest
-    sequences, single_end = _fastq_id_maker(per_sample_sequences=per_sample_sequences,
-                                            artifact_type=artifact_type)
-    barcode = 0
+    samples = per_sample_sequences.manifest.view(pd.DataFrame)
```
Comment (thermokarst) [inline q2_itsxpress/_itsxpress.py:182]: This invokes a transformer that loads the manifest as a pandas dataframe!

File: tests/test_main.py [hunk 9]
```diff
@@ -59,98 +65,57 @@
 ARTIFACT_TYPE_S = "SampleData[SequencesWithQuality]"
 
 
-def test_view_artifcat_type():
```
Comment (thermokarst) [inline tests/test_main.py:62]: This util was dropped, test no longer necessary.

File: tests/test_main.py [hunk 10]
```diff
@@ -59,98 +65,57 @@
...
-    exp1 = _itsxpress._view_artifact_type(per_sample_sequence=TEST_DATA)
-    eq_("SampleData[PairedEndSequencesWithQuality]", exp1)
-    raises(ValueError, lambda: _itsxpress._view_artifact_type(per_sample_sequence=TEST_DATA_PBMD))
-
-
-def test_write_metadata():
```
Comment (thermokarst) [inline tests/test_main.py:68]: This util was dropped, test no longer necessary.

File: tests/test_main.py [hunk 11]
```diff
@@ -59,98 +65,57 @@
...
-    results = SingleLanePerSampleSingleEndFastqDirFmt()
-    _itsxpress._write_metadata(results)
-    path = results.path
-    metadata = os.path.join(str(path), "metadata.yml")
-    with open(metadata, "rt") as fn:
-        eq_("{phred-offset: 33}", fn.readline().replace("\n", ""))
-
-
-def test_fastq_id_maker():
```
Comment (thermokarst) [inline tests/test_main.py:77]: This util was dropped, test no longer necessary.

File: tests/test_main.py [hunk 12]
```diff
@@ -59,98 +65,57 @@
...
-                                                single_end=True,
-                                                threads=1)
-        eq_(exp3[0], "4774-1-MSITS3")
-
-
-def test_trim_pair():
-    threads = 1
-    taxa = "F"
-    region = "ITS2"
+class SetFastqsAndCheckTests(unittest.TestCase):
```
Comment (thermokarst) [inline tests/test_main.py:73]: Please double-check me here, I am not 100% I captured the original intent of these tests.

File: tests/test_main.py [hunk 13]
```diff
@@ -177,18 +142,64 @@ def test_trim_pair_no_hmmer():
...
-                                                                    artifact_type=ARTIFACT_TYPE_P,
-                                                                    sequence=sequence,
-                                                                    single_end=single_end,
-                                                                    threads=1))
-        raises(ValueError, lambda: _itsxpress._set_fastqs_and_check(per_sample_sequences=TEST_DATA,
-                                                                    artifact_type=ARTIFACT_TYPE_S,
-                                                                    sequence=sequence,
-                                                                    single_end=single_end,
-                                                                    threads=1))
+class TrimTests(unittest.TestCase):
```
Comment (thermokarst) [inline tests/test_main.py:145]: These tests invoke QIIME 2 directly --- this not only tests the method functionality, but also ensures things like plugin registration is sound.

**Pull request merged**